landavailability/api/views.py

```diff
@@ -186,3 +188,25 @@ def post(self, request, format=None):
             return Response(serializer.data, status=status.HTTP_201_CREATED)
 
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def get(self, request, *args, **kwargs):
+        postcode = request.query_params.get('postcode')
+        range_distance = request.query_params.get('range_distance')
+
+        if postcode and range_distance:
+            try:
+                codepoint = CodePoint.objects.get(postcode=postcode)
+            except CodePoint.DoesNotExist as ex:
+                return Response(
+                    'The given postcode is not available in CodePoints',
+                    status=status.HTTP_400_BAD_REQUEST)
+
+            locations = Location.objects.filter(
```

Comment:
Exactly, the ranking will be done here. Being a custom API view we have quite enough freedom (as long as possibly we still have a recordset). So the idea would be to add a "rank" column to the recordset, calculate the ranking, assign the ranking and order by ranking.